### ISSUE TITLE
Issue loading Object Color

### ISSUE THREAD

**Issue by manikantagarikipati** [viromedia/virocore#72]
Hi,

I've started using it, Played around with the github examples, files with .vrx seems working fine.

Faced an issue loading .obj files but this [answer](https://github.com/viromedia/virocore/issues/2) helped me to render obj files.

But i bumped into a small issue regarding the color of the object which is rendered.

Sent an email with .mlt, .obj and textures files to support@viromedia.com

This is the screenshot where the color of chair is red rather than grey(according to the  textures provided).

![screenshot_20180406-123229_arandroid](https://user-images.githubusercontent.com/7972131/38408507-83108fdc-399b-11e8-8842-2a09e78b4c9f.jpg)

This is the code(AR placing objects sample in github except that its in Kotlin).
```
        mScene?.setListener(object: ARSceneListener(this, viroViewARCore){
            override fun onARInitialized() {
                trackingText.visibility = View.VISIBLE
                trackingText.setText(R.string.ar_is_intialised)
                addObject.visibility = View.VISIBLE
            }
        })
        // Add a light to the scene so our models show up
        mScene?.rootNode?.addLight(AmbientLight(Color.WHITE.toLong(), 1000f))

        viroViewARCore?.setScene(mScene)
```

And here is the 3DDraggableObject.kt

```
fun addModelToPosition(position: Vector,arScene: ARScene,type:Object3D.Type) {
        val object3D = Object3D()
        object3D.setPosition(position)
        // Shrink the objects as the original size is too large.
        object3D.setScale(Vector(.2f, .2f, .2f))
        object3D.gestureRotateListener = GestureRotateListener { i, node, rotation, rotateState ->
            if (rotateState == RotateState.ROTATE_START) {
                rotateStart = object3D.rotationEulerRealtime.y
            }
            val totalRotationY = rotateStart + rotation
            object3D.setRotation(Vector(0f, totalRotationY, 0f))
        }

        object3D.gesturePinchListener = GesturePinchListener { i, node, scale, pinchState ->
            if (pinchState == PinchState.PINCH_START) {
                scaleStart = object3D.scaleRealtime.x
            } else {
                object3D.setScale(Vector(scaleStart * scale, scaleStart * scale, scaleStart * scale))
            }
        }

        object3D.dragListener = DragListener { i, node, vector, vector1 -> }

        // Load the Android model asynchronously.
        object3D.loadModel(Uri.parse(mFileName), type, object : AsyncObject3DListener {
            override fun onObject3DLoaded(`object`: Object3D, type: Object3D.Type) {

            }

            override fun onObject3DFailed(s: String) {
                Toast.makeText(context, "An error occured when loading the 3D Object!",
                        Toast.LENGTH_LONG).show()
            }
        })

        // Make the object draggable.
        object3D.dragType = Node.DragType.FIXED_TO_WORLD
        arScene.rootNode.addChildNode(object3D)
    }
```

I'm calling the placeObject this way:

`placeObject("file:///android_asset/{myobject}.obj",Object3D.Type.OBJ)`

Am i missing anything here?

**Comment by dam00n** [comment]
Hi @manikantagarikipati, we took a look at the OBJ and the couch is actually red. Material color and illumination are controlled by the mtl file and the color was set as red. We modified the chair to grey (Kd 0.5000 0.5000 0.5000) but left the pillow as red (Kd 0.7804 0.0000 0.0000) by modifying the mtl. I will email that to you.

![img_0811](https://user-images.githubusercontent.com/450239/38427948-7be9c0d8-396f-11e8-9689-b05ab5e7209d.png)

Also, to see more details of the couch you might want to add a Directional or Omni light to your scene in addition to the Ambient light you already have.
